scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,13 +978,18 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                if (NodesPinger.pingNode(runningTask.getNodes().get(0)) == -1) {
+                    int attempts = runningTask.increaseAndGetPingAttempts();
+                    if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
```

Comment:
The property `PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS` is bypassed by your new ping attempts. If you don't try to restart immediately, is the restart behavior not changing? I mean is the failure counter (e.g. the one displayed in the Scheduler Web portal) still increasing? what happens if the node never reconnects? is it not waiting/retrying forever?